### PR TITLE
Feature/sw 5 08/35 jwt setup

### PR DESCRIPTION
**Pull request by JacobStokholms**
# Description
Integrating JWT to the current Login logic. 
Adding Controller templates for Teacher, Parent, Child, Admin. 
Adding BCrypt for password encryption

Fixes #\Feature/35

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Been tested through Swagger on a local database.

**Development Configuration**
CREATE DATABASE giraf

CREATE TABLE users (
id BIGSERIAL PRIMARY KEY NOT NULL,
first_name VARCHAR(30) NOT NULL,
last_name VARCHAR(100) NOT NULL,
email VARCHAR(100) NOT NULL,
password VARCHAR(100) NOT NULL,
role VARCHAR(15) NOT NULL,
status VARCHAR(20) NOT NULL
);

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [x] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have Acceptance Tested this on an iOS device
- [x] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: FoodplannerApi/Controller/AdminController.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+public class AdminController : BaseController
+{
+    private readonly UserService _userService;
+    
+    public AdminController(UserService userService){
+        _userService = userService;
+    }
+    
+    [HttpGet]
+    public async Task<IActionResult> GetAdminTest()
```
Comment (Androkless) [inline FoodplannerApi/Controller/AdminController.cs:18]: Var den her funktion noget i brugte til at teste med? 
Forstår i hvert fald ikke helt dens funktion 😄

File: FoodplannerApi/Controller/UsersController.cs [hunk 2]
```diff
@@ -7,66 +8,51 @@ namespace FoodplannerApi.Controller;
...
 
     [HttpPost]
-    public async Task<IActionResult> Create([FromBody] User user){
+    public async Task<IActionResult> Create([FromBody] UserCreateDTO userCreate){
         if (!ModelState.IsValid){
             return BadRequest(ModelState);
         }
-        var id = await _userService.CreateUserAsync(user);
+        var id = await _userService.CreateUserAsync(userCreate);
         if (id > 0){
```
Comment (Androkless) [inline FoodplannerApi/Controller/UsersController.cs:44]: Tjekker vi på om id'et allerede er oprettet et sted? Ellers skulle det måske tilføjes her

File: FoodplannerApi/Program.cs [hunk 3]
```diff
@@ -6,18 +8,62 @@
 using FoodplannerModels.Account;
 using FoodplannerServices;
 using FoodplannerServices.Account;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.IdentityModel.Tokens;
+using System.Text;
```
Comment (Androkless) [inline FoodplannerApi/Program.cs:17]: System.text er med 2 gange i denne fil

File: FoodplannerDataAccessSql/Account/UserRepository.cs [hunk 4]
```diff
@@ -55,17 +55,19 @@ public async Task<User> GetByIdAsync(int id)
```
Comment (Androkless) [inline FoodplannerDataAccessSql/Account/UserRepository.cs:55]: Kommentar til linje 50 der returnerer resultat:

Er det muligt at tjekke user dto'en vi returnerer - altså så vi ved felterne er udfyldt og ikke en null reference.

Er lidt usikker på hvordan validering foregår i .net 😄

File: FoodplannerApi/Controller/UsersController.cs (hunk 2 above)
Comment (JacobStokholms) [reply FoodplannerApi/Controller/UsersController.cs:44]: Alt logic skal ske i service layer, der ligger det også :)

File: FoodplannerDataAccessSql/Account/UserRepository.cs (hunk 4 above)
Comment (JacobStokholms) [reply FoodplannerDataAccessSql/Account/UserRepository.cs:55]: fixed

File: FoodplannerApi/Controller/AdminController.cs (hunk 1 above)
Comment (JacobStokholms) [reply FoodplannerApi/Controller/AdminController.cs:18]: Y i remove

**Review by alexrefshauge — APPROVED** [review]
Nice

**Pull request merged**